frontend/test/pytest/test_gradient.py

```diff
@@ -233,6 +233,57 @@ def f(x: float):
     assert np.allclose(result, expected)
 
 
+def test_value_and_grad_on_qjit_classical():
+    """Check that value_and_grad works when called on an qjit object that does not wrap a QNode."""
+
+    @qjit
+    def f(x: float):
+        return x * x
+
+    result = qjit(value_and_grad(f))(3.0)
+    expected = (9.0, 6.0)
+
+    assert np.allclose(result, expected)
+
+
+def test_value_and_grad_on_qjit_quantum():
+    """Check that value_and_grad works when called on an qjit object that does wrap a QNode."""
+
+    @qjit
+    def workflow(x: float):
+        @qml.qnode(qml.device("lightning.qubit", wires=3))
+        def circuit():
+            qml.CNOT(wires=[0, 1])
+            qml.RX(0, wires=[2])
+            return qml.probs()  # This is [1, 0, 0, ...]
```

Comment:
Hey @paul0403, sorry I didn't see this in advance before merging! I just wanted to check that this works as well when the input is an `n-dim` tensor? In this case, the output of `value_and_grad` should be

```python
(scalar, params_shape)
```

Reply:
I just tried and actually it doesn't; however, it's not an issue with `value_and_grad`, as `grad` has the same issue. I will open an issue tonight or tomorrow.

Reply:
https://github.com/PennyLaneAI/catalyst/issues/841